### PR TITLE
feat: `toBeExpanded` matcher

### PR DESCRIPTION
**Pull request by siepra**
### Summary

Part of Jest matchers migration
Following the example of https://github.com/callstack/react-native-testing-library/pull/1493

### Test plan

`yarn test`

### CODE REVIEW

**Comment by mdjastrzebski** [comment]
@siepra thanks for submitting this PR. During our experiments with iOS/Android RN runtime we've noticed that `expanded: false` state is behaving different than omitting the `expanded` entry. Detailed behavior here: https://github.com/callstack/react-native-testing-library/wiki/Accessibility:-State#expanded-state

Basically, RN treats explicit `expanded: false` as "collapsed" state, where as lack of `expanded` state means that this is neither expanded nor no collapsed. Therefore, it would make sense to introduce, in this PR, additional matcher of `toBeCollapsed` which would check `expanded === false` condition. Pls add `isElementCollapsed` as well + relevant tests.

File: src/helpers/accessiblity.ts [hunk 1]
```diff
@@ -226,3 +226,14 @@ export function isElementExpanded(
   const { accessibilityState, 'aria-expanded': ariaExpanded } = element.props;
   return ariaExpanded ?? accessibilityState?.expanded ?? false;
 }
+
+export function isElementCollapsed(
+  element: ReactTestInstance
+): NonNullable<AccessibilityState['expanded']> {
+  const { accessibilityState, 'aria-expanded': ariaExpanded } = element.props;
+  return ariaExpanded !== undefined
```
Comment (mdjastrzebski) [inline src/helpers/accessiblity.ts:234]: Why not `return (ariaExpanded ?? accessibilityState?.expanded) === false`

Comment (siepra) [reply src/helpers/accessiblity.ts:234]: It's the same, I just wanted to make sure the syntax will be correct for the default case. Changing it now

**Review by mdjastrzebski — APPROVED** [review]
Looks good! ✅

**Comment by mdjastrzebski** [comment]
🎉 Release in version: [v12.4.0](https://github.com/callstack/react-native-testing-library/releases/tag/v12.4.0)

**Pull request merged**